Validation/L1T/plugins/L1Validator.cc

```diff
@@ -82,78 +81,141 @@ void L1Validator::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetu
+
+  //For gen jet
+
+  for(uint i=0; i < GenJets->size(); i++){
+     const reco::GenJet *Genjet = &GenJets->at(i);
+     if(fabs(Genjet->eta())>4.7) continue;
+
+     double minDR = 999.0;
+     const l1t::Jet        *L1Part=NULL;
+     for(int iBx = JetsBX->getFirstBX();  iBx<=JetsBX->getLastBX(); ++iBx){
+          if(iBx>0) continue;
+          for(std::vector<l1t::Jet>::const_iterator jet = JetsBX->begin(iBx); jet != JetsBX->end(iBx); ++jet){
+                double idR = reco::deltaR(Genjet->eta(), Genjet->phi(), jet->eta(), jet->phi());
+                if( idR < minDR ){
+                         minDR = idR;
+                         L1Part = &(*jet);
+                }
+
+          }
+     }
+     _Hists.Fill(L1ValidatorHists::Type::Jet, Genjet, L1Part);
+  }
 
   for(uint i=0; i < GenParticles->size(); i++){
     const GenParticle *GenPart = &GenParticles->at(i);
 
     int pdg = GenPart->pdgId(), status = GenPart->status();
 
-    if(status==1 && (abs(pdg)==11 || pdg==22)){
-      FINDRECOPART(L1EmParticle, IsoEGs, NonIsoEGs)
-
-      if(RecoPart==NULL){
- 	_Hists.Fill(L1ValidatorHists::Type::IsoEG, GenPart, NULL);
- 	_Hists.Fill(L1ValidatorHists::Type::NonIsoEG, GenPart, NULL);
-      }else if(RecoPart->type() == L1EmParticle::EmType::kIsolated){
- 	_Hists.Fill(L1ValidatorHists::Type::IsoEG, GenPart, RecoPart);
- 	_Hists.Fill(L1ValidatorHists::Type::NonIsoEG, GenPart, NULL);
-      }else if(RecoPart->type() == L1EmParticle::EmType::kNonIsolated){
- 	_Hists.Fill(L1ValidatorHists::Type::IsoEG, GenPart, NULL);
- 	_Hists.Fill(L1ValidatorHists::Type::NonIsoEG, GenPart, RecoPart);
-      }
-    }else if(status==1 && abs(pdg)==13){
-      FINDRECOPART(L1MuonParticle, Muons, Muons)
-
-      _Hists.Fill(L1ValidatorHists::Type::Muon, GenPart, RecoPart);
-    }else if(status==3 && abs(pdg)==15){
-      FINDRECOPART(L1JetParticle, Taus, Taus)
-
-      _Hists.Fill(L1ValidatorHists::Type::TauJet, GenPart, RecoPart);
-    }else if(status==3 && (abs(pdg)<=5 || pdg==21)){
-      FINDRECOPART(L1JetParticle, CenJets, ForJets)
-
-      if(RecoPart==NULL){
- 	_Hists.Fill(L1ValidatorHists::Type::CenJet, GenPart, NULL);
- 	_Hists.Fill(L1ValidatorHists::Type::ForJet, GenPart, NULL);
-      }else if(RecoPart->type() == L1JetParticle::JetType::kCentral){
- 	_Hists.Fill(L1ValidatorHists::Type::CenJet, GenPart, RecoPart);
- 	_Hists.Fill(L1ValidatorHists::Type::ForJet, GenPart, NULL);
-      }else if(RecoPart->type() == L1JetParticle::JetType::kForward){
- 	_Hists.Fill(L1ValidatorHists::Type::CenJet, GenPart, NULL);
- 	_Hists.Fill(L1ValidatorHists::Type::ForJet, GenPart, RecoPart);
-      }
-    }else continue;
-
-    //cout << GenPart->pt() << '\t' << GenPart->eta() << '\t' << GenPart->phi() << '\t' << GenPart->pdgId() << endl;
+    double minDR = 999.0;
+    if(status==1 && abs(pdg)==13){  //Muon
+       if(fabs(GenPart->eta())>2.4) continue;
+       // calculate the isolation
+       double TotalIso30 = 0;
+       for(uint j=0; j<GenParticles->size(); j++){
```

Comment:
range-base for loop OK here, but for line 163, same as comment above, could stay with old style for loop?